### PR TITLE
API file structure refactoring & Implement frontend file login

### PR DESCRIPTION
**Pull request by sonowz**
Made `api` app in Django, and all other APIs will be managed there.
File login can be done in front page now, but still needs major information.
Solved issue #43.

Further noticeable changes are commented below.

### CODE REVIEW

File: backend/api/login/views.py [hunk 1]
```diff
@@ -28,15 +27,15 @@ class LoginRequest(APIView):
...
+    #   - Return sugang_list at success
     def post(self, request, option, *args, **kwargs):
         try:
             if option == 'mysnu':
                 logger.debug('mySNU login request with sessionid: ' + str(request.session.session_key))
                 user_id = request.data.get('user_id', 'nid')
                 password = request.data.get('password', 'npw')
                 sugang_list = mysnu.crawl_credit(user_id, password)
-                if len(sugang_list) == 0:
+                if sugang_list is None:
```
Comment (sonowz) [inline backend/api/login/views.py:38]: Code modified due to change in return value of `mysnu.crawl_credit`.

File: backend/api/login/views.py [hunk 2]
```diff
@@ -50,21 +49,23 @@ def post(self, request, option, *args, **kwargs):
                 text = text.decode('euc-kr', errors='ignore')
                 sugang_list = parse_credit(text)
 
+            if len(sugang_list) == 0:
+                raise ClientRenderedException('이수내역이 존재하지 않습니다.')
             rule = self.get_rule(request)
             # tree =
-            TreeLoader(rule, sugang_list, [])
+            TreeLoader(rule, sugang_list, None, [])
```
Comment (sonowz) [inline backend/api/login/views.py:56]: Code modified due to change in parameter of `TreeLoader` constructor.

File: backend/api/login/views.py [hunk 3]
```diff
@@ -50,21 +49,23 @@ def post(self, request, option, *args, **kwargs):
...
             # do something with 'tree' here
         # IOError should be handled within 'TreeLoader' later
         except IOError:
             logger.error('No such file: ' + rule)
             pass
         except ClientRenderedException as e:
             logger.error(e.args[0])
-            return HttpResponse(e.args[0])
+            return JsonResponse({'success': False, 'message': e.args[0]})
```
Comment (sonowz) [inline backend/api/login/views.py:64]: Modified due to change in API specification #28 .

File: frontend/src/elm/Pages/Prototype/FileSelectBox.elm [hunk 4]
```diff
@@ -1,36 +1,72 @@
...
 
-import Html exposing (Html, div, text, br, input)
-import Html.Attributes exposing (style, type_)
+import Html exposing (Html, div, text, br, input, button, form)
+import Html.Attributes exposing (style, type_, id, name, value)
+import Html.Events exposing (onClick, onSubmit)
+import Json.Decode as Json
+import Result exposing (Result(..))
+import Pages.Prototype.Ports as Ports
+import Pages.Prototype.Response as Response
```
Comment (sonowz) [inline frontend/src/elm/Pages/Prototype/FileSelectBox.elm:9]: `Pages.Prototype.Response` bundles repetitive JSON response decoding in one file.

File: frontend/src/elm/Pages/Prototype/LoginBox.elm [hunk 5]
```diff
@@ -58,7 +57,7 @@ update msg model =
     Submit ->
       let
         url =
-          URL.host ++ "login/mysnu/"
+          "/api/login/mysnu/"
```
Comment (sonowz) [inline frontend/src/elm/Pages/Prototype/LoginBox.elm:60]: No need to store hostname, using relative addressing.

File: frontend/src/elm/Utils/Http.elm [hunk 6]
```diff
@@ -1,19 +0,0 @@
-module Utils.Http exposing (..)
```
Comment (sonowz) [inline frontend/src/elm/Utils/Http.elm:1]: API response changed from `plain text` to `JSON`, so deleting unused functions.

File: frontend/src/static/index.js [hunk 7]
```diff
@@ -1,2 +1,4 @@
 Elm = require('../elm/main');
 Elm.Main.embed(document.getElementById('main'));
+
+fileRequest = require('../js/filerequest');
```
Comment (sonowz) [inline frontend/src/static/index.js:4]: Bundling `filerequest.js` into `main.js`.
I'm not sure whether it is properly handled.

Comment (dj-shin) [reply frontend/src/static/index.js:4]: Let's leave this verification to future testing work

**Review by dj-shin — APPROVED** [review]
Good. I guess with this we can close #43 and #28

**Pull request merged**